src/infrastructure/container.py

```
@@ -0,0 +1,96 @@
+from typing import Dict, Type, Any, Optional, TypeVar, Generic, Callable
+from functools import wraps
+import inspect
+from dataclasses import dataclass, field
```

Ruff (0.12.2)

`dataclasses.field` imported but unused

Remove unused import: `dataclasses.field`

(F401)